pkg/route/api/validation/validation.go

```diff
@@ -135,3 +140,35 @@ func validateNoDoubleEscapes(tls *routeapi.TLSConfig) fielderrors.ValidationErro
 	}
 	return allErrs
 }
+
+// validateTLSInsecure tests fields for different types of insecure options.
+// Called by validateTLS.
+func validateTLSInsecure(tls *routeapi.TLSConfig) *fielderrors.ValidationError {
+	// Check insecure option value if specified (empty is ok).
+	if tls == nil || tls.Insecure == "" {
+		return nil
+	}
+
+	// Ensure insecure is set only for edge terminated routes.
+	if routeapi.TLSTerminationEdge != tls.Termination {
+		// tls.Insecure option is not supported for a non edge-terminated routes.
+		msg := fmt.Sprintf("invalid value %s for insecure option - insecure is only supported for edge-terminated routes", tls.Insecure)
```

Comment:
way too verbose, the current value is already included... `fielderrors.NewFieldInvalid("insecure", tls.Insecure, "insecure is only allowed for edge-terminated routes")`